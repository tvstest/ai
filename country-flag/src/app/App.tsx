import React from 'react'
import { BrowserRouter, Switch, Route, Redirect } from 'react-router-dom'
import CountrySearch from 'app/pages/CountrySearch'
import CountryList from 'app/pages/CountryList'
import Navbar from 'app/components/Navbar'
import { Routes } from 'app/utils/enums/routes'
import { ThemeProvider } from '@mui/material'
import appTheme from 'app/configs/theme'

const App: React.FC = () => {
  return (
    <BrowserRouter>
      <ThemeProvider theme={appTheme}>
        <Navbar />
        <Switch>
          <Route exact path={Routes.Home} component={CountrySearch} />
          <Route exact path={Routes.CountriesByName} component={CountryList} />
          <Redirect to={Routes.Home} />
        </Switch>
      </ThemeProvider>
    </BrowserRouter>
  )
}
export default App
