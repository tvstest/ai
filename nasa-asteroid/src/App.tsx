import { ThemeProvider } from '@mui/material/styles'
import Navbar from 'app/components/Navbar'
import Routes from 'app/components/Routes'
import { Suspense } from 'react'
import theme from './app/config/theme'

const App: React.FC = () => {
  return (
    <Suspense fallback="Loading ...">
      <Navbar />
      <ThemeProvider theme={theme}>
        <Routes />
      </ThemeProvider>
    </Suspense>
  )
}

export default App
