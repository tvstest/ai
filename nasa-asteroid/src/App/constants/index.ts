export const NASA_BASE_URL = 'https://api.nasa.gov/neo/rest/v1/neo/'
export const NASA_API_KEY_PREFIX = `?api_key=${process.env.REACT_APP_NASA_API_KEY}`
