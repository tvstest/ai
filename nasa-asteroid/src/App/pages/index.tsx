import { Container, makeStyles } from "@material-ui/core";
import AsteroidDetails from "App/components/AsteroidDetails";
import AsteroidForm from "App/components/AsteroidForm";
import AsteroidContextWrapper from "App/context/AsteroidContext";

const useStyles = makeStyles({
  topMargin: {
    marginTop: 30,
  },
});

const Home = () => {
  const { topMargin } = useStyles();

  return (
    <Container className={topMargin}>
      <AsteroidContextWrapper>
        <AsteroidForm />
        <AsteroidDetails />
      </AsteroidContextWrapper>
    </Container>
  );
};

export default Home;
