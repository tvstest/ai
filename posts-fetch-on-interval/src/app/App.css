.App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

/* CSS Loader
--------------------------------------------------------------------------------------------------------*/
.loader-overlay {
  position: fixed;
  height: 100%;
  width: 100%;
  top: 0;
  left: 0;
  background: rgba(215, 215, 215, 0.35);
  z-index: 9999;
}
.cssload-fond {
  margin: auto;
  position: fixed;
  z-index: 0.5;
  top: 50%;
  left: 50%;
  margin: -20px 0 0 -20px;
}

.loaderHide {
  display: none;
}

.loaderShow {
  display: block;
}
