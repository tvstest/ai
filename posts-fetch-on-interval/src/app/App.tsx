import React, { Suspense } from 'react'
import { ThemeProvider } from '@mui/material/styles'
import { ToastContainer } from 'react-toastify'
import Loader from './components/Loader'
import Navbar from './components/Navbar'
import { theme } from './app.theme'
import 'react-toastify/dist/ReactToastify.css'
import './App.css'
import PostsData from './components/PostsData'

function App() {
  return (
    <Suspense fallback={<Loader />}>
      <ThemeProvider theme={theme}>
        <ToastContainer />
        <Navbar />
        <PostsData />
        <Loader />
      </ThemeProvider>
    </Suspense>
  )
}

export default App
