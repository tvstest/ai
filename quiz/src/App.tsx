import './App.css'
import { CircularProgress, ThemeProvider } from '@mui/material'
import { Suspense } from 'react'
import Routes from 'components/Routes'
import theme from 'configs/app-theme'
import { BrowserRouter as Router } from 'react-router-dom'

const App: React.FC = () => {
  return (
    <Suspense fallback={<CircularProgress />}>
      <ThemeProvider theme={theme}>
        <Router>
          <Routes />
        </Router>
      </ThemeProvider>
    </Suspense>
  )
}

export default App
