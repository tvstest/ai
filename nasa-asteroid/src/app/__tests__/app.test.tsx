import { render } from '@testing-library/react'
import { BrowserRouter as Router } from 'react-router-dom'
import App from 'App'

test('App loads with router and mui theme configuration properly', () => {
  render(
    <Router>
      <App />
    </Router>
  )
})

test('matches the app snapshot', async () => {
  expect(
    <Router>
      <App />
    </Router>
  ).toMatchSnapshot()
})
