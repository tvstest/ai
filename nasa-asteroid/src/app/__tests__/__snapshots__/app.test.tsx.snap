// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the app snapshot 1`] = `
<BrowserRouter>
  <App />
</BrowserRouter>
`;
